### PR TITLE
Cleanup attribute handlers

### PR DESCRIPTION
**Pull request by nox**
This makes all tests in attributes.html pass.

### CODE REVIEW

**Comment by nox** [comment]
The third commit probably (that I just reworded) probably breaks Element::removeAttribute() but the edge case I'm thinking about is missing a test.

**Comment by nox** [comment]
I broke Critic rewording the third commit.

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -610,6 +610,8 @@ pub trait AttributeHandlers {
     /// name, if any.
     fn get_attribute(self, namespace: Namespace, local_name: &Atom)
                      -> Option<Temporary<Attr>>;
+    /// Returns the first attribute with given case-sensitive name, if any.
```
Comment (Manishearth) [inline components/script/dom/element.rs]: nit: I'd prefer if the docs explicitly mentioned that it returns the first attr with a given case sensitive name and any namespace

File: components/script/dom/element.rs [hunk 2]
```diff
@@ -748,17 +759,14 @@ impl<'a> AttributeHandlers for JSRef<'a, Element> {
...
     }
 
-    fn remove_attribute(self, namespace: Namespace, name: &str) {
-        let (_, local_name) = get_attribute_parts(name);
-        let local_name = Atom::from_slice(local_name);
-
-        let idx = self.attrs.borrow().iter().map(|attr| attr.root()).position(|attr| {
-            *attr.r().local_name() == local_name
-        });
+    fn remove_attribute<F>(self, cb: F) where F: Fn(JSRef<Attr>) -> bool {
```
Comment (Manishearth) [inline components/script/dom/element.rs]: nit: let's give the argument a name that's a tad more descriptive (`filter`?)

File: components/script/dom/element.rs [hunk 3]
```diff
@@ -625,7 +627,7 @@ pub trait AttributeHandlers {
     fn parse_attribute(self, namespace: &Namespace, local_name: &Atom,
                        value: DOMString) -> AttrValue;
 
-    fn remove_attribute(self, namespace: Namespace, name: &str);
+    fn remove_attribute<F>(self, cb: F) where F: Fn(JSRef<Attr>) -> bool;
```
Comment (Manishearth) [inline components/script/dom/element.rs]: docs: "removes the first attribute that ... "

File: components/script/dom/element.rs [hunk 4]
```diff
@@ -1071,16 +1081,19 @@ impl<'a> ElementMethods for JSRef<'a, Element> {
...
 
     // http://dom.spec.whatwg.org/#dom-element-removeattributens
     fn RemoveAttributeNS(self,
                          namespace: Option<DOMString>,
                          localname: DOMString) {
-        let namespace = namespace::from_domstring(namespace);
-        self.remove_attribute(namespace, &localname)
+        let namespace = &namespace::from_domstring(namespace);
+        let localname = &Atom::from_slice(&localname);
+        self.remove_attribute(|attr| {
```
Comment (Manishearth) [inline components/script/dom/element.rs]: I think a `remove_attribute_ns(ns, name)` would be useful to have in these cases (internally just delegates to `remove_attribute()` with the correct filter)

**Comment by nox** [comment]
@Manishearth Good to go.
